src/mnode/mnode-messageproc.h

```diff
@@ -13,21 +13,30 @@ extern CCriticalSection cs_mapOurMessages;
 bool Sign(const std::string& message, std::string& signatureBase64, std::string& error_ret);
 bool Sign(const std::string& message, std::vector<unsigned char>& signature, std::string& error_ret);
 
+// Type to distinguish the way we build/parse messages.
+enum class CMasternodeMessageType
+{
+    PLAINTEXT,
+    SETFEE
+};
+
 class CMasternodeMessage
 {
 public:
     CTxIn vinMasternodeFrom;
     CTxIn vinMasternodeTo;
+    int messageType;
     std::string message;
     int64_t sigTime{}; //message times
     std::vector<unsigned char> vchSig;
 
     CMasternodeMessage() = default;
     
-    CMasternodeMessage(COutPoint outpointMasternodeFrom, COutPoint outpointMasternodeTo, const std::string& msg) :
+    CMasternodeMessage(COutPoint outpointMasternodeFrom, COutPoint outpointMasternodeTo, CMasternodeMessageType msgType, const std::string& msg) :
         vinMasternodeFrom(outpointMasternodeFrom),
         vinMasternodeTo(outpointMasternodeTo),
         sigTime(0),
+        messageType(static_cast<int>(msgType)),
```

Comment:
static_cast< unsigned short >

Reply:
why unsigned short? the default underlying type of an enum class is always int

Reply:
To save database space. You can use even uint8_t